### DISCUSSION TITLE
Allow altering the opacity of the hitobjects on the waveform timeline in the editor

### DISCUSSION THREAD

**Discussion by Hydria** [ppy/osu#29325] [Ideas]
![](https://hydria.s-ul.eu/wyOWAh8m)
if i want to see if this object is on time or not, i have to physically move the object, which, if i'm modding a mapset, i want to avoid doing so as much as possible, so having an option to adjust object opacity means we can check if objects are snapped correctly without having to physically alter the map

example of what 50% opacity looks like:
![](https://hydria.s-ul.eu/QJzP9Cc2)

**Comment by bdach** [comment]
I'm confused.

What are you wanting to correlate against what?

- If you want to make sure the beatmap is properly timed (e.g. the timing points are correctly placed), use the timing tab which has no objects.
- If you want to make sure the objects are snapped to the beat, use the verify tab.

What am I missing here?

**Reply by Hydria** [reply]
comparing the waveform to the hitobject to see if they align (for songs that have notes which don't always typically align on standard beats)

**Reply by peppy** [reply]
A workaround may be to switch to timing view momentarily (F3 -> F1).

**Reply by Hydria** [reply]
that can work for the time being, but a more permanent solution would be preferred in this case when there's free time to do so, no rush

**Comment by asmorodinov** [comment]
Hello, I made a PR addressing this discussion, but it was rejected - https://github.com/ppy/osu/pull/35135
(I have searched for existing issues and PR-s, but I think I forgot to search for a discussion before making it)

So, I wanted to ask, what other option can be implemented instead? Peppy suggested adding a toggle or hold-to-hide button, is it ok if I implement it? Or what else can be done?

By the way, there is kind of similar discussion about transparency, and in stable timeline objects are already transparent by default (I only use lazer, so I didn't know about that) - https://github.com/ppy/osu/discussions/21999.
And this one as well - https://github.com/ppy/osu/discussions/21229 (third point).

---

P.S. I apologize for making a PR too early without looking for discussions and also spamming in the PR comments. I thought this was a trivial change that was easy to implement, so I made it relatively quickly.

For me, this issue is already solved, since I can use my locally built lazer client, but I wanted to help other mappers as well.
